src/Microsoft.DotNet.Interactive.Formatting/Formatter.cs

```diff
@@ -145,20 +172,35 @@ public static void SetPreferredMimeTypeFor(Type type, string preferredMimeType)
                 throw new ArgumentException("Value cannot be null or whitespace.", nameof(preferredMimeType));
             }
 
+
             ClearComputedState();
-            
-            _preferredMimeTypes.Push((type, preferredMimeType));
+
+            _preferredDataMimeTypes.Push((type, preferredMimeType));
         }
 
-        public static string DefaultMimeType
+        public static string DefaultDisplayMimeType
         {
-            get => _defaultMimeType;
-            set => _defaultMimeType = value ??
+            get => _defaultDisplayMimeType;
+            set => _defaultDisplayMimeType = value ??
                                       throw new ArgumentNullException(nameof(value));
         }
 
-        public static string GetPreferredMimeTypeFor(Type type) =>
-            _preferredMimeTypesTable.GetOrAdd(type ??= typeof(object), InferMimeType);
+        public static string DefaultDataMimeType
+        {
+            get => _defaultDataMimeType;
+            set => _defaultDataMimeType = value ??
+                                          throw new ArgumentNullException(nameof(value));
+        }
+
+        public static IReadOnlyCollection<string> GetPreferredDisplayMimeTypesFor(Type type) =>
+            _preferredDisplayMimeTypesTable.GetOrAdd(
+                type ??= typeof(object), 
+                key => InferMimeTypes(key, _preferredDisplayMimeTypes, _defaultPreferredDisplayMimeTypes, _defaultDisplayMimeType));
+
+        public static string GetPreferredDataMimeTypeFor(Type type) =>
```

Comment:
I don't think splitting this into two methods makes things clearer. If I'm understanding the change, this keeps the `FirstOrDefault` ambiguity in `GetPreferredDisplayMimeTypesFor` and introduces a new ambiguity around which of these methods to use at any given time.

My thought about "display" versus "data" MIME types that we discussed wasn't intended to be reflected in the API. I was just trying to draw attention to the idea that when multiple are registered, they have different purposes. For example, say both `text/html` and `application/json` are registered. We can choose between them based on the context. Are we displaying something? We could use `text/html`. Are we transferring data? We could use `application/json`.